metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py

```diff
@@ -300,6 +301,18 @@ class SnowflakeV2Config(
         "to ignore the temporary staging tables created by known ETL tools.",
     )
 
+    table_name_normalization_rules: Dict[re.Pattern, str] = pydantic.Field(
+        default={},
+        description="[Advanced] Regex patterns for table names to normalize in lineage ingestion. "
```

Comment:
does it make sense for us to have `table_name_normalization_rules` and `extra_table_name_normalization_rules` - setting the former overrides the defaults, but the latter lets you extend it?

the ruff linter uses a similar pattern which works pretty well https://docs.astral.sh/ruff/settings/#lint_extend-select

Reply:
That's an excellent idea. Have not injected defaults yet. That's exactly the problem I was looking to solve - how to let users extend defaults without re-declaring them, and at the same time, allow users to change defaults if they do not suit them.